security/tls-1.3.md

```diff
@@ -0,0 +1,219 @@
+configuration, TLS version negotiation is left to server and the client.
+
+## Components
+
+### BMO
+
+**controller-manager**:
+
+- Port `8443`: TLS 1.2, TLS 1.3
+
+TLS version is not configurable with flags.
+
+**ironic**:
+
+For Ironic, it should be noted that:
+
+- Ironic external endpoints are secured, while pod-internal traffic is not.
+  `httpd` (Apache) handles TLS termination.
+- IPA image serving from `httpd` port `6180` might be limited to insecure due
+  PXE limitations (to be verified)
+- Node image server is deployment specific. IPA client accessing the image
+  server is limited to TLS 1.2 (see Oslo below)
+
+Ports:
+
+- Port `5049`: HTTP (Ironic Inspector API)
+- Port `5050`: TLS 1.2, TLS 1.3 (httpd - Inspector endpoint)
+- Port `6180`: HTTP (httpd - serving IPA images)
+- Port `6385`: TLS 1.2, TLS 1.3 (httpd - Ironic endpoint)
+- Port `6388`: HTTP (Ironic API)
+
+Ironic endpoints support setting minimum and maximum TLS versions.
+
+More info:
+
+- [OSLO library does not support](https://docs.openstack.org/oslo.service/latest/configuration/index.html#ssl.version)
+  TLS newer than 1.2 yet, TLS 1.3 support is planned
+- [IPA TLS configuration documentation](https://docs.openstack.org//ironic-python-agent/latest/doc-ironic-python-agent.pdf)
+
+### CAPI
+
+All CAPI controllers support setting minimum and maximum TLS versions, with the
+exception of `CAPD` which is a test provider.
+
+**capi-kubeadm-bootstrap**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+**capi-kubeadm-control-plane**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+**capi-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+### CAPM3
+
+**capm3-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+TLS version is not configurable with flags.
+
+**ipam-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+TLS version is not configurable with flags.
+
+### cert-manager
+
+**cert-manager**:
+
+- Port `9402`: HTTP (metrics)
+
+**cert-manager-cainjector**:
+
+- no listening ports
+
+**cert-manager-webhook**:
+
+- Port `10250`: TLS 1.2, TLS 1.3
+- Port `6080`: HTTP (healthz)
+
+### kube-system
+
+**coredns**:
+
+- Port `53` TCP/UDP: DNS
+- Port `9153`: HTTP (metrics)
+
+**etcd**:
+
+For etcd, TLS 1.3 support was
+[contributed by EST](https://github.com/etcd-io/etcd/pull/15156) and is expected
+to be in the next release of etcd 3.5 (March 2023). In etcd versions from `3.5`
+to `3.5.7` have hardcoded TLS 1.2 version. Some versions of etcd `3.4` have
+support for TLS 1.3 as they lack this hardcoded TLS version, and using new
+enough Golang enables TLS 1.3 for them.
+
+`etcd` will support setting minimum and maximum TLS versions after the PR is merged.
```

Comment:
link to the pr?

Reply:
PR linked in the above line where the patch is discussed.